k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -83,19 +84,19 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
    *
    * TODO: investigate k8s websockets support as an alternative HTTP approach
    *
-   * @return An asynchronous stream of [[Watch]] objects containing additions, modifications, and
-   *         deletions to the items in the list being watched, and a Closable handle allowing for
-   *         the watch to be terminated.
+   * @return An update function that may be passed to [[Var.async]] to create stream of [[Watch]]
+   *         objects containing additions, modifications, and deletions to the items in the list
+   *         being watched, and a Closable handle allowing for the watch to be terminated.
    */
   def watch(
     labelSelector: Option[String] = None,
     fieldSelector: Option[String] = None,
     resourceVersion: Option[String] = None
-  ): (AsyncStream[W], Closable) = {
+  ): (Activity.State[W] => Unit) => Closable = { (state: Activity.State[W] => Unit) =>
```

Comment:
Could this return type just be `Activity[W]` instead?  I think this should be as simple as moving `Activity(Var.async(...) { ... })` from the caller into this method.  That would make this method signature a little easier to understand.

Reply:
IIRC I actually started this change with watch returning a Event[].  It was indeed cleaner, and worked in deployment, but the tests failed:
- some of the existing tests write multiple events into the response to be processed before actually testing what came out on the other end.
- the only way to consume a stream of events from a Var or Activity seems to be Var.changes or Activity.states (which then calls Var.changes)
- in the observe method of the Var created in Var.async, I *think* multiple updates were managing to sneak in between the synchronized creation of the nested Var and the call to .observe which would register the Observer, so the first update would not necessarily be seen
- and so the tests would not necessarily see the first event and would fail.

I thought this behavior of missing some initial events should be ok in the wild (we'd get the latest event eventually), but wanted to make sure the tests (that test the `watch` function directly) saw the complete stream of events, as they did previously with AsyncStream.  I was also trying to avoid changing the logic of the tests.  So I ended up with this function  thus this function and the concurrent-queue machinery in the tests. 

I don't see a way of extracting the complete stream of events out of a Var (otherwise we'd likely run into the same issue as with AsyncStream, right?).  Anything I may be missing about Vars/Activities?

Actually... another option to improve the readability without changing the signature much (obvious in retrospect):
```
def watch(
    labelSelector: Option[String] = None,
    fieldSelector: Option[String] = None,
    resourceVersion: Option[String] = None,
    state: Activity.State[W] => Unit
  ) 
```
The watchable call-site doesn't actually benefit from being able to partially watch, and the ApiTest callsite is easily changed to work with this signature as well.

Thoughts?   It'd still keep the weird `state` type but make the `watch` signature plainer, and the tests can still avoid dealing with Var quirks.

Reply:
per discussion, went with the l updated signature